pytroll_collectors/tests/test_segments.py

```diff
@@ -329,134 +325,152 @@ def test_get_collection_status(self):
         mda = self.mda_msg0deg_south_segment.copy()
         msg = FakeMessage(mda)
         col = self.msg0deg_north
-        with self._caplog.at_level(logging.DEBUG):
+        with caplog.at_level(logging.DEBUG):
             col.process(msg)
-            logs = [rec.message for rec in self._caplog.records]
+            logs = [rec.message for rec in caplog.records]
             assert "H-000-MSG3__-MSG3________-VIS006___-000008___-201611281100-__ not in " in logs[2]
 
-    def test_process_message_twice(self):
+    def test_process_message_twice(self, caplog):
         """Test processing a message."""
         mda = self.mda_msg0deg.copy()
         msg = FakeMessage(mda)
         col = self.msg0deg
         message = Message(msg, col._patterns['msg'])
         col._create_slot(message)
         col.process(msg)
-        with self._caplog.at_level(logging.DEBUG):
+        with caplog.at_level(logging.DEBUG):
             col.process(msg)
-            logs = [rec.message for rec in self._caplog.records]
+            logs = [rec.message for rec in caplog.records]
             assert 'File already received' in logs
 
-    def test_process_message_unknown_file(self):
+    def test_process_message_unknown_file(self, caplog):
         """Test processing a message with an unknown file uid."""
         mda = self.mda_msg0deg.copy()
         mda['uid'] = "blablabla"
         msg = FakeMessage(mda)
         col = self.msg0deg
-        with self._caplog.at_level(logging.DEBUG):
+        with caplog.at_level(logging.DEBUG):
             col.process(msg)
-            logs = [rec.message for rec in self._caplog.records]
+            logs = [rec.message for rec in caplog.records]
             assert 'No parser matching message, skipping.' in logs
 
+    def test_process_message_force_local_files(self):
+        """Test processing message with scheme when config says files are local."""
+        mda = self.mda_msg0deg.copy()
+        expected_uri = mda['uri']
+        mda['uri'] = 'SCHEME://host.foo.bar' + mda['uri']
+        msg = FakeMessage(mda)
+        col = SegmentGatherer(CONFIG_SINGLE)
+        col._config['all_files_are_local'] = True
+        col.process(msg)
+        slot = col.slots[str(mda['start_time'])]
+        uri = slot.output_metadata['dataset'][0]['uri']
+        assert uri == expected_uri
+
+    def test_process_message_force_local_files_check_existing(self):
+        """Test processing message with scheme when config says files are local and existing files are checked."""
+        mda = self.mda_msg0deg.copy()
+        mda['uri'] = 'SCHEME://host.foo.bar' + mda['uri']
+        msg = FakeMessage(mda)
+        col = SegmentGatherer(CONFIG_SINGLE)
+        # Without forcing all files to be local the processing would fail with
+        # "ValueError: Protocol not known: scheme"
+        col._config['all_files_are_local'] = True
+        col._config['check_existing_files_after_start'] = True
+        col.process(msg)
```

Comment:
Do we need to add an assert here for something?

Reply:
Without the handling added in this PR the test will crash, so I thought just that the test passes would be enough. If you have an idea what else to check, I'm all ears.

Reply:
that's all good :)